gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemIntegrationTest.java

```diff
@@ -651,8 +649,91 @@ private void testFileChecksum(
     assertThat(fileChecksum.getAlgorithmName()).isEqualTo(checksumType.getAlgorithmName());
     assertThat(fileChecksum.getLength()).isEqualTo(checksumType.getByteLength());
     assertThat(fileChecksum.getBytes()).isEqualTo(checksumFn.apply(fileContent));
+    assertThat(fileChecksum.toString()).contains("%s: ".format(checksumType.getAlgorithmName()));
 
     // Cleanup.
-    assertThat(ghfs.delete(filePath, true)).isTrue();
+    assertThat(ghfs.delete(filePath, /* recursive= */ true)).isTrue();
+  }
+
+  @Test
+  public void testInitializeWithEmptyWorkingDirectory() throws IOException {
+    GoogleHadoopFileSystem myGhfs = (GoogleHadoopFileSystem) ghfs;
+    Configuration config = loadConfig();
+    ghfs.initialize(myGhfs.initUri, config);
+    String rootBucketName = myGhfs.getRootBucketName();
+    config.unset(GoogleHadoopFileSystemConfiguration.GCS_WORKING_DIRECTORY.getKey());
+    ghfs.initialize(myGhfs.initUri, config);
+
+    assertThat(ghfs.getHomeDirectory().toString()).startsWith("gs://" + rootBucketName);
+  }
+
+  @Test
+  public void testGlobStatusOptions() throws IOException {
+    testGlobStatusFlatConcurrent(true, true);
+    testGlobStatusFlatConcurrent(true, false);
+    testGlobStatusFlatConcurrent(false, true);
+    testGlobStatusFlatConcurrent(false, false);
+  }
+
+  private void testGlobStatusFlatConcurrent(boolean flat, boolean concurent) throws IOException {
+    Configuration configuration = getConfigurationWtihImplementation();
+    configuration.setBoolean(
+        GoogleHadoopFileSystemConfiguration.GCS_FLAT_GLOB_ENABLE.getKey(), flat);
+    configuration.setBoolean(
+        GoogleHadoopFileSystemConfiguration.GCS_CONCURRENT_GLOB_ENABLE.getKey(), concurent);
+
+    ghfs.initialize(ghfs.getUri(), configuration);
+    Path testRoot = new Path("/directory1/");
+    ghfs.mkdirs(testRoot);
+    ghfs.mkdirs(new Path("/directory1/subdirectory1"));
+    createFile(
+        new Path("/directory1/subdirectory1/file1"), "data".getBytes(StandardCharsets.UTF_8));
+    FileStatus[] rootDirectories = ghfs.globStatus(new Path("/d*"));
+
+    assertThat(rootDirectories).hasLength(1);
```

Comment:
No need to check number of records, if exact content already asserted in next assertion.